### DISCUSSION TITLE
Custom Component for Nav Items

### DISCUSSION THREAD

**Discussion by MinSomai** [payloadcms/payload#9339] [Feature Requests & Ideas]
Current issue is that I want to put icons in the sidebar nav for each collection.

Having custom component to replace each items in the side nav and ability to edit grups too maybe.

Looks like I already created this in the past: Duplicate: https://github.com/payloadcms/payload/discussions/6768#discussioncomment-11499694

**Comment by yonnic** [comment]
yea the sidebar really need some attention. Its currently too small and totally "naked". Hope the whole thing can be replaced with a custom component soon

**Comment by auduongtuan** [comment]
I'm in the same need too 😭 Currently using CSS to achieve this.

**Comment by bratvanov** [comment]
These are some of the improvements I'd like to see to Payload's admin panel nav component:

- Support for multi-level nesting in navigation groups with configurable visibility (always shown or expandable/collapsible with a chevron). Currently, navigation only goes 1 level deep.
- Allow specifying an SVG icon for nav items via a string path or inline SVG code.
- Enable adding standalone links to the navigation without associating them with a collection or global (e.g for pointing to custom admin routes with custom views).

**Comment by MinSomai** [comment]
this repo has custom icons example, thanks to the dev

https://github.com/akhrarovsaid/payload-theme-quantum-leap

**Reply by MinSomai** [reply]
this is interesting too, with plugin

https://github.com/tatsuyakari1203/payload-sidebar

**Reply by MinSomai** [reply]
another interesting one
https://github.com/VeiaG/payload-enhanced-sidebar

**Comment by Mhmod-Hsn** [comment]
I've built a simple plugin to achieve what you need, 
[https://github.com/Mhmod-Hsn/Payload-Advanced-Sidebar](https://github.com/Mhmod-Hsn/Payload-Advanced-Sidebar
)
most welcome for improvements
